### PR TITLE
Feature: APP-2402 - Token Transfers For Base Networks

### PR DESCRIPTION
**Pull request by Fabricevladimir**
## Description

- adds support for token transfer fetching for Base mainnet and testnet
- adds feature flags for networks in DAO creation

Note: Please review _after_ #958 since its branch is the base branch for this PR

Task: [APP-2402](https://aragonassociation.atlassian.net/browse/APP-2402)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them if necessary.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [x] I have tested my code on the test network.

[APP-2402]: https://aragonassociation.atlassian.net/browse/APP-2402?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/services/token/domain/alchemy-token.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+export type AlchemyTransfer = {
```
Comment (cgero-eth) [inline src/services/token/domain/alchemy-token.ts]: File name should be `alchemy-transfer.ts`

File: src/services/token/domain/covalent-token.ts [hunk 2]
```diff
@@ -34,3 +34,25 @@ export type CovalentTokenBalance = {
...
 };
+
+export type CovalentTokenTransfer = {
+  address: string;
+  items: Array<{
+    transfers: CovalentTransferInfo[];
+  }>;
+};
+
+export type CovalentTransferInfo = {
```
Comment (cgero-eth) [inline src/services/token/domain/covalent-token.ts]: I'd move these two types on a new `covalent-transfer.ts` file, what do you think?

File: src/services/token/query-keys.ts [hunk 3]
```diff
@@ -6,20 +6,22 @@ import type {
...
   ],
-};
-
-export const tokenBalancesQueryKeys = {
-  address: (params: IFetchTokenBalancesParams): QueryKey => [
+  balances: (params: IFetchTokenBalancesParams): QueryKey => [
     TokenQueryItem.BALANCES,
     params,
   ],
+  transfers: (params: IFetchTokenBalancesParams): QueryKey => [
```
Comment (cgero-eth) [inline src/services/token/query-keys.ts]:
```suggestion
  transfers: (params: IFetchTokenTransfersParams): QueryKey => [
```

File: src/services/token/token-service.ts [hunk 4]
```diff
@@ -217,6 +228,116 @@ class TokenService {
...
+
+    // check if network is supported
+    if (!networkId) {
+      console.info(
+        `fetchCovalentErc20Deposits - network ${network} not supported by Covalent`
+      );
+      return null;
+    }
+    // fetch all balances
+    const assetsBalances = await this.fetchTokenBalances({
```
Comment (cgero-eth) [inline src/services/token/token-service.ts]: By making these api calls here we are not using the react-query cache, this means that we will be calling the `fetch-token-balance` api regardless if we already fetched on the UI. We could fix this by passing the token balances as a parameter to this function.

File: src/hooks/useDaoTransfers.tsx [hunk 5]
```diff
@@ -1,192 +1,131 @@
-import {useReactiveVar} from '@apollo/client';
```
Comment (cgero-eth) [inline src/hooks/useDaoTransfers.tsx:1]: This hook looks much cleaner now, well done!

**Review by cgero-eth — COMMENTED** [review]
Thanks for the changes Fabrice!

File: src/services/token/domain/covalent-token.ts (hunk 2 above)
Comment (Fabricevladimir) [reply src/services/token/domain/covalent-token.ts]: Might as well, yup

File: src/services/token/token-service.ts (hunk 4 above)
Comment (Fabricevladimir) [reply src/services/token/token-service.ts]: Good catch 👍🏾

**Review by cgero-eth — APPROVED** [review]

**Pull request merged**